packages/wdio-spec-reporter/src/index.ts

```diff
@@ -140,7 +145,10 @@ export default class SpecReporter extends WDIOReporter {
                 ? '.us-east-1'
                 : ['eu', 'eu-central-1'].includes(config.region || '') ? '.eu-central-1' : ''
             const multiremoteNote = isMultiremote ? ` ${instanceName}` : ''
-            return [`Check out${multiremoteNote} job at https://app${dc}.saucelabs.com/tests/${sessionId}`]
+            const sauceLabsSharableLinks = this._sauceLabsSharableLinks
+                ? sauceAuthenticationToken( <string>config.user, <string>config.key, <string>sessionId )
```

Comment:
in order to avoid these castings I would suggest to do this:

```js
if (isSauceJob && config.user && config.key && sessionId) {
```

Because they could technically be undefined.

Reply:
Yep, that worked, will update it